### ISSUE TITLE
Render MarkupText colors using <span> tags

### ISSUE THREAD

**Issue by MrMallIronmaker** [ManimCommunity/manim#1039]
Currently, coloring uses `<color>` tags which are Manim-specific (unsupported by Pango), and were a workaround for lack of color parsing in the SVG engine. Now that SVG parsing for fill and color has improved (#915), MarkupText can pass a foreground color to Pango, and Pango's "official" coloring tag, `<span foreground="#rrggbb">`, can be used.

As discussed later in #915, there are two parts:
1. Passing in the fill color (which fill color?) to Pango as the `AttrForeground` attribute.
2. Removing the `init_colors` override for MarkupText objects.

cc @PhilippImhof

**Comment by PhilippImhof** [comment]
So now I have changed `MarkupText` in the following ways:

* pass `f'<span foreground="{self.color}">{self.text}</span>'` instead of just `self.text` to Pango in order to set the default color for all chars
* completely remove `def init_colors(...)` 

The SVG file is correctly created, e.g. all chars that are not in a particular `<span foreground="...">` are set in the default color (white in my case) and custom coloring of individual chars is correctly applied (gray background added manually just to make the white chars visible):
![image](https://user-images.githubusercontent.com/52650214/108620947-d291fd80-742f-11eb-91ab-5ca55da2147a.png)

However, the text does not appear in the animation:
![image](https://user-images.githubusercontent.com/52650214/108620959-e9d0eb00-742f-11eb-962f-a571ca22b041.png)



This is with the latest commits from PR #915. Is there still something on the SVG end that has to be changed?

**Comment by MrMallIronmaker** [comment]
What line exactly is the call to Pango? I tried 1287 but it gave me an attribute error - looks like self.color is unset at this point in the process. If I manually tell it to be white, then I do reproduce your bug.

I think this has to do with #1041, which requires a little bit of work on my end. I'm in the middle of it now.

**Comment by MrMallIronmaker** [comment]
Make the two changes (fill color + init_colors) to MarkupText on this branch - https://github.com/MrMallIronmaker/manim/tree/better-use

I was able to get the correct rendering that way, and included a test with an already-rendered version of the Pango MarkupText output. I haven't written a test for the MarkupText class yet, though.

**Comment by PhilippImhof** [comment]
> What line exactly is the call to Pango? I tried 1287 but it gave me an attribute error - looks like self.color is unset at this point in the process. If I manually tell it to be white, then I do reproduce your bug.

`self.color` used to be unset, because there was no need for it; I am setting it now in `MarkupText`'s `__init__`. Pango comes into play at the very end of the `text2svg` function. However, IMHO there is currently nothing to do over there, because the SVG is correctly created, e.g. with white (or whatever one chooses) as the default color.

> I think this has to do with #1041, which requires a little bit of work on my end. I'm in the middle of it now.

I'll wait for you to finish before pulling PR #915 again and (re-)doing my updates. I am just doing it in order to prepare the changes and have them ready once #915 is merged. OTOH, it makes no sense at all to actually create a PR with these changes before.

**Comment by PhilippImhof** [comment]
> Make the two changes (fill color + init_colors) to MarkupText on this branch - https://github.com/MrMallIronmaker/manim/tree/better-use

I'd rather just wait for you to finish your work on #915 and use this as the base as soon as it's merged. If I start a PR on your repo and we integrate that into your (already huge) PR, things just get more and more complicated IMHO. There's no need to hurry anyway, because all the functionality is already there; it will just become cleaner better for text with ligatures.

**Comment by MrMallIronmaker** [comment]
True true - only need to clone if you want to double-check the behavior is
correct - definitely no need for a PR.

**Comment by PhilippImhof** [comment]
I have just retried using the most recent commits from #915. The SVG is again correctly created (e.g. white for uncolored chars), but the text does not show up. This is with `init_colors` removed from the `MarkupText` class.

**Comment by MrMallIronmaker** [comment]
It's likely the same issue as before - I didn't make any changes to #915 with regards to this bug. Sorry for the confusion. Do you think I should merge my better-use branch with #915?

**Comment by PhilippImhof** [comment]
Ah, then it was just a misunderstanding. I do not know how many changes that would make, you have already put in so much effort. On one hand, it would be good to have it fixed directly with the (hopefully soon-to-come) merge of #915, because I could then just file a PR to close this issue. (Instead of a joint PR or two separate PRs later on.) OTOH, it means more work for you and the reviewers, unless it's just a few lines of code.

**Comment by MrMallIronmaker** [comment]
It's a decent amout, unfortunately (https://github.com/MrMallIronmaker/manim/compare/master...MrMallIronmaker:better-use)

> showing 5 changed files with 115 additions and 39 deletions

Is there some way we can do a merge behind the scenes, and ultimately only make one PR to the community edition?

**Comment by PhilippImhof** [comment]
OMG, that's way too much to ask. It is probably the best if you add another PR later on. I should then be able to push commits to that PR inside this repo. That's certainly the easiest way to go, especially because (as I suppose) there are not many changes for the `MarkupText` part.

**Comment by MrMallIronmaker** [comment]
Will do. I can either make the PR now and rebase when #915 is merged, or wait until #915 is merged to make the PR. Do you have a preference?

**Comment by PhilippImhof** [comment]
I'd say your preference will be mine; you are the one doing the tough job :)

**Comment by MrMallIronmaker** [comment]
PR #1058 has been opened!
